internal/core/server/workspace.go

```diff
@@ -599,6 +647,7 @@ func (ws *Workspace) RenameComponent(ctx context.Context, input *api.RenameCompo
 
 func (ws *Workspace) RefreshComponents(ctx context.Context, input *api.RefreshComponentsInput) (*api.RefreshComponentsOutput, error) {
 	query := makeComponentQuery(withRefs(input.Refs...))
+	// TODO: Recompute dependencies on refresh.
```

Comment:
I'm not quite clear why. Is this because we ignore a failure to get dependencies on apply?

Reply:
I'll expand the comment. The main reason is because the analysis for dependencies in theory may be out of date, either because the provider implementation changed or because it depends on some non-deterministic data (none do currently), so the idea is that "refresh" should fixup and busted state as best it can.